lib/console/ai/tools/alerts.ex

```diff
@@ -0,0 +1,85 @@
+defmodule Console.AI.Tools.Alerts do
+  use Ecto.Schema
+  import Ecto.Changeset
+  import Ecto.Query
+  import Console.AI.Tools.Utils
+  alias Console.Schema.{Alert, Flow}
+
+  embedded_schema do
+    field :severities, {:array, :string}
+    field :state,      :string
+    field :types,       {:array, :string}
+  end
+
+  @valid ~w(severities state types)a
+
+  def changeset(model, attrs) do
+    model
+    |> cast(attrs, @valid)
+  end
+
+  @json_schema Console.priv_file!("tools/alerts.json") |> Jason.decode!()
+
+  def json_schema(), do: @json_schema
+  def name(), do: plrl_tool("alerts")
+  def description(), do: "Shows alerts for the current flow, with optional filtering by severity, state, and type."
+
+  def implement(%__MODULE__{severities: severities, state: state, types: types}) do
+    case Console.AI.Tool.flow() do
+      %Flow{id: flow_id} ->
+        query_builder(Alert, severities, state, types)
+        |> Alert.for_flow(flow_id)
+        |> Console.Repo.all()
+        |> model()
+        |> Jason.encode()
+      nil ->
+        {:error, "no flow found"}
+    end
+  end
+
+  def query_builder(base_query, severities, state, types) do
+    [
+      {severities, :apply_severity_filter},
+      {state,      :apply_state_filter},
+      {types,      :apply_type_filter}
+    ]
+    |> Enum.reduce(base_query, fn {value, filter_fun_atom}, current_query ->
+      apply(__MODULE__, filter_fun_atom, [current_query, value])
+    end)
+  end
+
+  def apply_severity_filter(query, [_ | _] = severities) do
+    atomic_severities = Enum.map(severities, &String.to_atom/1)
+    from(q in query, where: q.severity in ^atomic_severities)
```

Comment:
don't manually define the queries here, instead put them in the alert schema module and call those.  It makes it easier to reuse and organize the queries against the db.